src/sentry/static/sentry/app/components/events/interfaces/threads/threadsSelectorOption.tsx

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import styled from '@emotion/styled';
+
+import space from 'app/styles/space';
+import TextOverflow from 'app/components/textOverflow';
+import InlineSvg from 'app/components/inlineSvg';
+import {EntryTypeData} from 'app/types';
+
+interface Props {
+  id: string;
+  name?: string;
+  crashed?: boolean;
+  details: ThreadInfo;
+  crashedInfo?: EntryTypeData;
+}
+
+interface ThreadInfo {
+  label: string;
+  filename?: string;
+}
+
+const ThreadsSelectorOption: React.FC<Props> = ({
+  id,
+  name,
+  details,
+  crashed,
+  crashedInfo,
+}) => (
+  <StyledContainer>
+    <StyledNameID>{name ? `#${id}: ${name}` : `#${id}`}</StyledNameID>
+    <StyledLabelsContainer>
+      <StyledOptionLabel>{details.label}</StyledOptionLabel>
+      {details.filename && (
+        <StyledFileName>
+          {'('}
+          <StyledOptionLabel>{details.filename}</StyledOptionLabel>
+          {')'}
+        </StyledFileName>
+      )}
+    </StyledLabelsContainer>
+    {crashed && (
+      <StyledCrashIcon
+        src="icon-warning-sm"
+        title={crashedInfo ? `(crashed with ${crashedInfo.values[0].type})` : ''}
+      />
+    )}
+  </StyledContainer>
+);
+
+export default ThreadsSelectorOption;
+
+const StyledLabelsContainer = styled('div')({
```

Comment:
I think all of these styled components are named very well and don't need the Styled prefix